scripts/breaking_changes_checker/breaking_changes_tracker.py

```diff
@@ -559,25 +560,43 @@ def check_module_level_function_removed_or_renamed(self, function_components: Di
                 )
             return True
 
-    # ----------------------------------- Report methods -----------------------------------
     def get_reportable_breaking_changes(self, ignore_changes: Dict) -> List:
-        reportable_changes = []
-        ignored = ignore_changes[self.package_name]
-        for bc in self.breaking_changes:
-            msg, bc_type, module_name, *args = bc
-            class_name = args[0] if args else None
-            function_name = args[1] if len(args) > 1 else None
-            if (bc_type, module_name) in ignored or \
-                    (bc_type, module_name, class_name) in ignored or \
-                    (bc_type, module_name, class_name, function_name) in ignored:
-                continue
-            reportable_changes.append(bc)
-        return reportable_changes
+        ignored = []
+        # Match all ignore rules that should apply to this package
+        for ignored_package, ignore_rules in ignore_changes.items():
+            if re.findall(ignored_package, self.package_name):
+                ignored.extend(ignore_rules)
+        # Remove ignored breaking changes from list of reportable changes
+        for ignored_bc, *ignored_args in ignored:
+            ignored_module_name = ignored_args[0] if ignored_args else None
+            ignored_class_name = ignored_args[1] if len(ignored_args) > 1 else None
+            ignored_function_name = ignored_args[2] if len(ignored_args) > 2 else None
+            ignored_parameter_name = ignored_args[3] if len(ignored_args) > 3 else None
+            for bc in self.breaking_changes:
+                _, bc_type, module_name, *args = bc
+                class_name = args[0] if args else None
+                function_name = args[1] if len(args) > 1 else None
+                parameter_name = args[2] if len(args) > 2 else None
+            
+                # If the ignore item is a wildcard, it should match all values
+                if ignored_module_name == "*":
+                    module_name = "*"
+                if ignored_class_name == "*":
+                    class_name = "*"
+                if ignored_function_name == "*":
+                    function_name = "*"
```

Comment:
I think we need to reset these with every iteration of the loop

Reply:
Arent we doing that on lines 577-579?